lib/pengine/complex.c

```diff
@@ -876,28 +876,6 @@ pe__unpack_resource(xmlNode *xml_obj, pe_resource_t **rsc,
     return pcmk_rc_ok;
 }
 
-void
-common_update_score(pe_resource_t * rsc, const char *id, int score)
-{
-    pe_node_t *node = NULL;
-
-    node = pe_hash_table_lookup(rsc->allowed_nodes, id);
-    if (node != NULL) {
-        pe_rsc_trace(rsc, "Updating score for %s on %s: %d + %d", rsc->id, id, node->weight, score);
-        node->weight = pcmk__add_scores(node->weight, score);
-    }
-
-    if (rsc->children) {
-        GList *gIter = rsc->children;
-
-        for (; gIter != NULL; gIter = gIter->next) {
-            pe_resource_t *child_rsc = (pe_resource_t *) gIter->data;
-
-            common_update_score(child_rsc, id, score);
```

Comment:
Previously this was recursive. Just to confirm, we're still fine now, because `ban_unavailable_allowed_nodes()` is only called with an instance, which can only have one level of children (clone -> group (instance) -> members) -- right?

Reply:
Exactly